src/main/java/edu/ie3/datamodel/io/source/sql/SqlIdCoordinateSource.java

```diff
@@ -0,0 +1,215 @@
+public class SqlIdCoordinateSource extends SqlDataSource<CoordinateValue>
+    implements IdCoordinateSource {
+  private static final String WHERE = " WHERE ";
+  /**
+   * Queries that are available within this source. Motivation to have them as field value is to
+   * avoid creating a new string each time, bc they're always the same.
+   */
+  private final String basicQuery;
+
+  private final String queryForPoint;
+  private final String queryForPoints;
+  private final String queryForId;
+  private final String queryForBoundingBox;
+
+  /**
+   * Initializes a new SqlIdCoordinateSource
+   *
+   * @param connector the connector needed for the database connection
+   * @param schemaName the database schema to use
+   * @param coordinateTableName the name of the table containing coordinate data
+   * @param factory instance of a coordinate factory
+   */
+  public SqlIdCoordinateSource(
+      SqlConnector connector,
+      String schemaName,
+      String coordinateTableName,
+      IdCoordinateFactory factory) {
+    super(connector);
+
+    String dbIdColumnName = getDbColumnName(factory.getIdField(), coordinateTableName);
+    String dbPointColumnName = getDbColumnName(factory.getPointField(), coordinateTableName);
+
+    // setup queries
+    this.basicQuery =
+        createBaseGisQueryString(
+            schemaName, coordinateTableName, dbIdColumnName, dbPointColumnName);
+    this.queryForPoint = createQueryForPoint(dbIdColumnName);
+    this.queryForPoints = createQueryForPoints(dbIdColumnName);
+    this.queryForId = createQueryForId(dbPointColumnName);
+    this.queryForBoundingBox = createQueryForBoundingBox(dbPointColumnName);
+  }
+
+  @Override
+  protected Optional<CoordinateValue> createEntity(Map<String, String> fieldToValues) {
+    int id;
+    Point point;
+
+    try {
+      id = Integer.parseInt(fieldToValues.get("id"));
+      double longitude = Double.parseDouble(fieldToValues.get("stX"));
+      double latitude = Double.parseDouble(fieldToValues.get("stY"));
+
+      point = GeoUtils.buildPoint(latitude, longitude);
+    } catch (Exception e) {
+      return Optional.empty();
+    }
+
+    return Optional.of(new CoordinateValue(id, point));
+  }
+
+  @Override
+  public Optional<Point> getCoordinate(int id) {
+    List<CoordinateValue> values = executeQuery(queryForPoint, ps -> ps.setInt(1, id));
+
+    return Optional.of(values.get(0).coordinate);
+  }
+
+  @Override
+  public Collection<Point> getCoordinates(int... ids) {
+    Object[] idSet = Arrays.stream(ids).boxed().distinct().toArray();
+
+    List<CoordinateValue> values =
+        executeQuery(
+            queryForPoints,
+            ps -> {
+              Array sqlArray = ps.getConnection().createArrayOf("int", idSet);
+              ps.setArray(1, sqlArray);
+            });
+
+    List<Point> points = new ArrayList<>();
+
+    for (CoordinateValue value : values) {
+      points.add(value.coordinate);
+    }
+
+    return points;
```

Comment:
This can be simplified. Please check in this file for further similar occurrences.

```suggestion
    return values.stream().map(value -> value.coordinate).toList();
```